nucypher/blockchain/eth/token.py

```diff
@@ -163,7 +170,7 @@ def __init__(self,
 
         # Agency
         self.staking_agent = None
-        self.token_agent = NucypherTokenAgent()
+        self.token_agent = NucypherTokenAgent()  # TODO: Use Agency
```

Comment:
Do you really need to use the Agency? Isn't this way good enough as the objects are supposed to be singletons automagically, due to Agency being the metaclass?

Reply:
The function signature of agent requires the `blockchain` parameter, so except for the *first* instantiation, it can be misleading to do:
- `StakingEscrowAgent(blockchain=blockchain)` (will not use this blockchain) 
- `StakingEscrowAgent()` (missing the required arg) 

As an alternate API - I've started using `Agency.get_agent(NucypherTokenAgent)` - perhaps the blockchain instance used for agent instances can be better managed there, too. 

What do you think about this pattern?